src/main/java/com/basho/riak/client/core/query/timeseries/SetCell.java

```diff
@@ -0,0 +1,59 @@
+package com.basho.riak.client.core.query.timeseries;
+
+import com.basho.riak.client.api.convert.Converter;
+import com.basho.riak.client.api.convert.ConverterFactory;
+import com.basho.riak.client.core.query.Namespace;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.client.core.util.Constants;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.Set;
+
+public class SetCell extends Cell
+{
+    private static final Type StringType = new TypeReference<String>() {}.getType();
+    private static final Namespace GenericNamespace = new Namespace("T");
+    private static final BinaryValue GenericValue = BinaryValue.create("");
+
+    public static <T> SetCell fromSet(Set<T> value, TypeReference<T> typeReference)
+    {
+        SetCell cell = new SetCell();
```

Comment:
We may change the signature of https://github.com/basho/riak-java-client/pull/543/files#diff-4e432a46681bf2317f99d9c32411a94dR212 to return SetCell